pkg/adapter/adapter.go

```diff
@@ -281,31 +282,51 @@ func (a *Adapter) PollForMessages(channel *wabbit.Channel,
 
 	msgs, _ := a.ConsumeMessages(channel, queue, logger)
 
+	wg := &sync.WaitGroup{}
+	workerCount := a.config.ChannelConfig.PrefetchCount
```

Comment:
Prefetch counts of 100 or even 1000 are common recommendations for high-throughput scenarios. Are we sure that we want to start that many goroutines?

Is there going to be a single source per go process? If we are likely to have many sources per go process, this may cause concurrency problems - locking & context switching seem too high to me.

Reply:
What do you mean by "sources"? In general, Goroutines are lightweight and having thousands of them does not necessarily impact (thread) context switching. Not sure where we'd have locking bottlenecks in this code path.